src/orders/orders.repository.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { FilterQuery, Model } from 'mongoose';
+import { Order, OrderDocument } from './schemas/order.schema';
+import { getAllOrdersResponse } from './schemas/order.types';
+
+@Injectable()
+export class OrdersRepository {
+  constructor(
+    @InjectModel(Order.name) private orderModel: Model<OrderDocument>,
+  ) {}
+
+  async findAllById(
+    orderFilterQuery: FilterQuery<OrderDocument>,
+  ): Promise<Order[]> {
+    const { id, sortBy } = orderFilterQuery.query;
+
+    return this.orderModel.find({ userId: id }).sort([['date', sortBy]]);
+  }
+
+  async find(
+    orderFilterQuery: FilterQuery<OrderDocument>,
+  ): Promise<getAllOrdersResponse> {
+    const { page, limit, sortBy, sortRow } = orderFilterQuery.query;
+    const total = await this.orderModel.count();
+
+    const items = await this.orderModel
```

Comment:
DRY: you implemented the same method for both Order and User entities. Does it makes sense to create some helpers where you would pass specific model (orderModel, userModel) and do the same, your method turns into something like that:
```
  async find(orderFilterQuery: FilterQuery<OrderDocument>) {
    return helper. orderFilterQuery(this.orderModel, orderFilterQuery)
 }
```